### PR TITLE
Add godoc for the core stats types

### PR DESCRIPTION
**Pull request by rakyll**
We will need more comprehensive docs but I am planning to write language agnostic specs first and reuse the content there.

Also remove NewTestingCountAggregationValue which is trivial to implement.

### CODE REVIEW

File: stats/view.go [hunk 1]
```diff
@@ -71,19 +72,19 @@ func NewView(name, description string, keys []tags.Key, measure Measure, agg Agg
...
 }
 
-// Name returns the name of view.
+// Name returns the name of the view.
 func (v *View) Name() string {
 	return v.name
 }
 
-// Description returns the name of view.
+// Description returns the name of thte view.
```
Comment (bogdandrutu) [inline stats/view.go]: s/thte/the?

File: plugins/grpc/stats/server_handler_test.go [hunk 2]
```diff
@@ -357,3 +357,8 @@ func TestServerDefaultCollections(t *testing.T) {
 		}
 	}
 }
+
+func countAggregationValue(v int) *istats.CountAggregationValue {
```
Comment (bogdandrutu) [inline plugins/grpc/stats/server_handler_test.go]: shouldn't this be "newCountAggregationValue" based on the code style?

File: stats/aggregation.go [hunk 3]
```diff
@@ -15,13 +15,18 @@
 
 package stats
 
-// Aggregation is the generic interface for all aggregtion types.
+// Aggregation represents a data aggregation method. There are several
+// aggregation methods made available by the package such as
```
Comment (bogdandrutu) [inline stats/aggregation.go]: s/by the package/in the package?

File: stats/measure.go [hunk 4]
```diff
@@ -15,10 +15,21 @@
...
+// Each measure needs to be registered before being used.
+// Measure constructors such as NewMeasureInt64 and
+// NewMeasureFloat64 automatically registers the measure
+// by the given name.
+// Each registered measure needs to be unique by name.
+// Measures also have units that represent the unit of the
+// metric they are representing.
 type Measure interface {
 	Name() string
+	Unit() string
```
Comment (bogdandrutu) [inline stats/measure.go]: I think this has to be a different PR. This supposed to be doc PR but also changes functionality. Please remove this change from this PR.

File: plugins/grpc/stats/server_handler_test.go (hunk 2 above)
Comment (rakyll) [reply plugins/grpc/stats/server_handler_test.go]: Done

File: stats/aggregation.go (hunk 3 above)
Comment (rakyll) [reply stats/aggregation.go]: Done

File: stats/measure.go (hunk 4 above)
Comment (rakyll) [reply stats/measure.go]: Agreed, removed.

File: stats/view.go (hunk 1 above)
Comment (rakyll) [reply stats/view.go]: Done

**Comment by rakyll** [comment]
Actually, let me revert the changes to EqualRows as well before you take another look.

**Comment by rakyll** [comment]
PTAL
